File: src/main/java/com/github/wakingrufus/mastodon/ui/feeds/NotificationController.java
package com.github.wakingrufus.mastodon.ui.feeds;

import com.github.wakingrufus.mastodon.ui.ViewAccountKt;
import com.sys1yagi.mastodon4j.api.entity.Notification;
import javafx.fxml.FXML;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class NotificationController {
    private final Notification notification;
    @FXML
    Label displayName;
    @FXML
    Label notificationTime;
    @FXML
    Label message;
    @FXML
    HBox accountView;
    @FXML
    HBox tootView;

    public NotificationController(Notification notification) {
        this.notification = notification;
    }

    @FXML
    public void initialize() {
        displayName.setText(notification.getAccount().getDisplayName());
        if (notification.getType().equalsIgnoreCase("follow")) {
            message.setText("followed you.");
            ViewAccountKt.viewAccount(accountView, notification.getAccount());
        } else if (notification.getType().equalsIgnoreCase("favourite")) {
            message.setText("has favourited your status: " + notification.getStatus().getContent());

        } else {
            message.setText(notification.getType());
        }
        notificationTime.setText(notification.getCreatedAt());
    }
}

File: src/main/kotlin/com/github/wakingrufus/mastodon/ui/controllers/NotificationController.kt
package com.github.wakingrufus.mastodon.ui.controllers

import com.github.wakingrufus.mastodon.account.AccountState
import com.github.wakingrufus.mastodon.ui.Viewer
import com.github.wakingrufus.mastodon.ui.ViewerMode
import com.sys1yagi.mastodon4j.api.entity.Account
import com.sys1yagi.mastodon4j.api.entity.Notification
import com.sys1yagi.mastodon4j.api.entity.Status
import javafx.fxml.FXML
import javafx.scene.control.Label
import javafx.scene.layout.HBox
import mu.KLogging


class NotificationController(private val notification: Notification,
                             private val accountPrompter: () -> AccountState?,
                             private val statusViewer: Viewer<Status> = Viewer(
                                     controller = { item -> TootController(status = item, accountPrompter = accountPrompter) },
                                     template = "/toot.fxml"),
                             private val accountViewer: Viewer<Account> = Viewer(
                                     controller = { account -> AccountController(account) },
                                     template = "/account.fxml")) {
    companion object : KLogging()

    @FXML
    internal var displayName: Label? = null
    @FXML
    internal var notificationTime: Label? = null
    @FXML
    internal var message: Label? = null
    @FXML
    internal var accountView: HBox? = null
    @FXML
    internal var tootView: HBox? = null

    @FXML
    fun initialize() {
        displayName?.text = notification.account!!.displayName
        if (notification.type.equals("follow", ignoreCase = true)) {
            message?.text = "followed you."
            accountViewer.view(accountView!!, notification.account!!)
        } else if (notification.type.equals("favourite", ignoreCase = true)) {
            message?.text = "has favourited your status"
            statusViewer.view(parent = tootView!!, item = notification.status!!, mode = ViewerMode.APPEND)
        } else {
            message?.text = notification.type
        }
        notificationTime?.text = notification.createdAt
    }
}